### PR TITLE
Support reporter to report scalar variable

### LINKED ISSUE #1439: Is it possible to merge two different input layers into one?
Keras provide [Merge Layer](http://keras.io/layers/core/#merge) for this, is it possible in chainer?

### LINKED ISSUE #1459: Cannot report int value
I couldn't report int value. We need to check type of an input value here:
https://github.com/pfnet/chainer/blob/master/chainer/reporter.py#L244

### PR DESCRIPTION
**Pull request by unnonouno**
Fix #1459

### CODE REVIEW

File: chainer/reporter.py [hunk 1]
```diff
@@ -221,6 +221,13 @@ def report_scope(observation):
     current.observation = old
 
 
+def _get_device(x):
```
Comment (delta2323) [inline chainer/reporter.py]: Is it difficult to integrate this function to original `get_device` method?

Comment (unnonouno) [inline chainer/reporter.py]: The original `get_device` treat an int value as a device number.

Comment (delta2323) [inline chainer/reporter.py]: I see. Thanks

**Comment by delta2323** [comment]
LGTM 👍

**Pull request merged**